src/components/DragAndDrop.js

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+
+import variables from '../styles/variables';
+
+const COPY_DROP_EFFECT = 'copy';
+const NONE_DROP_EFFECT = 'none';
+
+const DragAndDropCallbackPropTypes = {
+    /** Callback to fire when a file has being dragged over the text input & report body */
+    onDragOver: PropTypes.func.isRequired,
+
+    /** Callback to fire when a file has been dragged into the text input & report body */
+    onDragEnter: PropTypes.func.isRequired,
+
+    /** Callback to fire when the user is no longer dragging over the text input & report body */
+    onDragLeave: PropTypes.func.isRequired,
+
+    /** Callback to fire when a file is dropped on the text input & report body */
+    onDrop: PropTypes.func.isRequired,
+};
+
+const propTypes = {
+    /** Rendered child component */
+    children: PropTypes.node.isRequired,
+
+    /** Rendered child component */
+    dropZoneId: PropTypes.string.isRequired,
+
+    ...DragAndDropCallbackPropTypes,
+};
+
+export default class DragAndDrop extends React.Component {
```

Comment:
Please refactor this component to make it platform-safe. I won't scope-creep and require that you add any drag-and-drop support on native platforms, but you should create a directory structure like so:

```
----| src/components/DragAndDrop
--------| index.js // Here is the web implementation
--------| index.native.js // This can just be a no-op that renders children unchanged. 
```